gsy_framework/sim_results/electric_blue/volume_timeseries.py

```diff
@@ -0,0 +1,160 @@
+        energy_kWh = asset_stats.total_energy_bought
+        if energy_kWh <= 0:
+            return
+        profile = self._trade_profile_generator.generate_trade_profile(
+            energy_kWh, asset_stats.time_slot, market_type)
+        for time_slot, value in profile.items():
+            self._add_to_volume_time_series(time_slot, value, market_type, "bought_kWh")
+
+    def _add_total_energy_sold(
+            self, asset_stats: ForwardDeviceStats, market_type: AvailableMarketTypes):
+        """Add statistics for the total amount of sold energy."""
+        energy_kWh = asset_stats.total_energy_sold
+        if energy_kWh <= 0:
+            return
+        profile = self._trade_profile_generator.generate_trade_profile(
+            energy_kWh, asset_stats.time_slot, market_type
+        )
+        for time_slot, value in profile.items():
+            self._add_to_volume_time_series(time_slot, value, market_type, "sold_kWh")
+
+    def _add_to_volume_time_series(
+            self, time_slot: DateTime, value: float,
+            market_type: AvailableMarketTypes, attribute_name: str):
+        """Add time slot value to the correct time slot to asset volume time series."""
+        time_slot = self._adapt_time_slot(time_slot)
+        volume_time_series = self._get_asset_volume_time_series(time_slot.start_of("year"))
+        volume_time_series[time_slot][market_type.name][attribute_name] += value
+
+    def _adapt_time_slot(self, time_slot: DateTime) -> DateTime:
+        """Find the containing time slot of the smaller time slot."""
+        if self.resolution == AggregationResolution.RES_15_MINUTES:
+            return time_slot.set(minute=(time_slot.minute // 15) * 15)
+        return time_slot.start_of(START_OF[self.resolution])
+
+    def _get_asset_volume_time_series(self, time_slot):
+        """Return asset volume time series for the required time slot.
+        If not found in the buffer, it tries fetching it from DB.
+        If not found in the DB, it will generate a new one for the whole year."""
+        if time_slot in self._asset_volume_time_series_buffer:
+            time_series = self._asset_volume_time_series_buffer[time_slot]
+        else:
+            time_series = self._fetch_asset_volume_time_series_from_db(time_slot)
+            if time_series is None:
+                time_series = {
+                    ts: self._get_time_series_template(value)
+                    for ts, value in self._generate_asset_volume_time_series(time_slot)}
+            self._asset_volume_time_series_buffer[time_slot] = time_series
+        return time_series
+
+    def _fetch_asset_volume_time_series_from_db(
+            self, time_slot: DateTime) -> Optional[Dict[DateTime, Dict]]:
+        """Fetch already saved asset volume time series."""
+        # TODO: should be implemented.
+        return None
+
+    def _generate_asset_volume_time_series(self, time_slot: DateTime):
```

Comment:
AFAIU, this method "only" creates a time series in the self.resolution that contains the SSP values. It does not return the volume time_series, right? So I suggest to rename the method (to e.g. `generate_SSP_time_series`) and adapt the docstring.

Reply:
Fixed.